tests/indexes/tests.py

```diff
@@ -139,6 +166,33 @@ def test_ops_class_multiple_columns(self):
             )
             self.assertCountEqual(cursor.fetchall(), expected_ops_classes)
 
+    def test_ops_class_partial(self):
+        index = Index(
+            name='test_ops_class_partial',
+            fields=['body'],
+            opclasses=['text_pattern_ops'],
+            condition=Q(headline__contains='China'),
+        )
+        with connection.schema_editor() as editor:
+            editor.add_index(IndexedArticle2, index)
+        with editor.connection.cursor() as cursor:
+            cursor.execute(self.get_opclass_query % 'test_ops_class_partial')
+            self.assertCountEqual(cursor.fetchall(), [('text_pattern_ops', 'test_ops_class_partial')])
+
+    def test_ops_class_partial_tablespace(self):
+        index = Index(
+            name='test_ops_class_tblspace',
+            fields=['body'],
+            opclasses=['text_pattern_ops'],
+            condition=Q(headline__contains='China'),
+            db_tablespace='pg_default',
+        )
+        with connection.schema_editor() as editor:
+            editor.add_index(IndexedArticle2, index)
+        with editor.connection.cursor() as cursor:
+            cursor.execute(self.get_opclass_query % 'test_ops_class_tblspace')
+            self.assertCountEqual(cursor.fetchall(), [('text_pattern_ops', 'test_ops_class_tblspace')])
```

Comment:
I don't know much about the tablespace feature, but I'd expect to see an assertion that interacts with it somehow.

Reply:
Good point. I was mainly interested in seeing that the index is created, as the tablespace string is inserted before the condition. As far as I can see, `pg_default` is not in the `pg_indexes` table and `indexdef` does not contain the `TABLESPACE ...` string, so I'm just testing that the generated SQL contains it.